pkg/daemon/daemon.go

```diff
@@ -149,24 +149,30 @@ func NewDaemon() (Daemon, error) {
 		return nil, err
 	}
 
-	// Reset guid pool with already allocated guids to avoid collisions
-	err = syncGUIDPool(smClient, guidPool)
+	// Create daemon instance first to access syncGuidPool method
+	daemonInstance := &daemon{
+		config:            daemonConfig,
+		watcher:           nil, // Will be set later
+		kubeClient:        client,
+		guidPool:          guidPool,
+		smClient:          smClient,
+		guidPodNetworkMap: make(map[string]string),
+	}
+
+	// Initialize guid pool with existing pods and sync with subnet manager
+	err = daemonInstance.initGUIDPool()
```

Comment:
I suggest to not run it here anymore since we only should start the actual logic if the instance becomes a leader. In the `becomeLeader` function we are already running `initGUIDPool`

Reply:
Let's refactor this function to return the daemon fully formed at the end

Reply:
done.